AstSemantics.md

```diff
@@ -231,6 +239,40 @@ operator may be added. However, due to normal fragmentation, applications are
 instead expected release unused physical pages from the working set using the
 [`discard`](FutureFeatures.md#finer-grained-control-over-memory) future feature.
 
+## Tables
+
+A *table* is similar to a linear memory whose elements, instead of being bytes,
+are opaque values of a particular *table element type*. This allows the table to
+contain values like GC references, raw OS handles, or native pointers that are
+accessed by WebAssembly code indirectly through an integer index. This feature
+bridges the gap between low-level, untrusted linear memory and high-level
+opaque handles/references at the cost of a bounds-checked table indirection.
+
+The table's element type dynamically constrains the type of elements stored 
+in the table and allows engines to avoid some type checks on table use. When
+tables are mutated, any stored value is necessarily coerced (possibly trapping)
+to the element type.
```

Comment:
@jfbastien Expanded and clarified.  The idea is that, in the future if we have `set_table`, we could ensure the element types match statically.  But when mutating through the dynamic JS API (or other host APIs), some sort of dynamic check or coercion would be necessary.